src/services/inplayer.account.service.ts

```diff
@@ -44,36 +80,170 @@ export const logout = async () => {
+    };
+  } catch {
+    throw new Error('Failed to update user data.');
+  }
+};
+
+export const getPublisherConsents: GetPublisherConsents = async (config) => {
+  try {
+    const { inplayer } = config.integrations;
+    const { data } = await InPlayer.Account.getRegisterFields(inplayer?.clientId || '');
+
+    // @ts-ignore
+    // wrong data type from InPlayer SDK (will be updated in the SDK)
+    const result: Array<Consent> = data?.collection
+      .filter((field: GetRegisterField) => field.type === 'checkbox')
+      .map((consent: GetRegisterField) => processPublisherConsents(consent));
+
+    return {
+      errors: [],
+      responseData: {
+        consents: [getTermsConsent(), ...result],
+      },
+    };
+  } catch {
+    throw new Error('Failed to fetch publisher consents.');
+  }
+};
+
+export const getCustomerConsents = async (payload: GetCustomerConsentsPayload) => {
+  try {
+    const { customer } = payload;
+
+    if (!customer?.metadata) {
+      return {
+        errors: [],
+        responseData: {
+          consents: [],
+        },
+      };
+    }
+
+    const consents = JSON.parse((customer.metadata?.consents as string) || '');
+
+    return {
+      errors: [],
+      responseData: {
+        consents,
+      },
+    };
+  } catch {
+    throw new Error('Unable to fetch Customer consents.');
+  }
+};
+
+export const updateCustomerConsents: UpdateCustomerConsents = async (payload) => {
+  try {
+    const { customer, consents } = payload;
+
+    const data = {
+      consents,
+      firstName: customer.metadata?.firstName as string,
+      lastName: customer.metadata?.lastName as string,
+    };
+
+    return (await updateCustomer(data, true, '')) as ServiceResponse<never>;
+  } catch {
+    throw new Error('Unable to update Customer`s consents');
+  }
+};
+
+export const getCaptureStatus: GetCaptureStatus = async ({ customer }) => {
+  return {
+    errors: [],
+    responseData: {
+      isCaptureEnabled: true,
+      shouldCaptureBeDisplayed: true,
+      settings: [
+        {
+          answer: {
+            firstName: customer.firstName || null,
+            lastName: customer.lastName || null,
+          },
+          enabled: true,
+          key: 'firstNameLastName',
+          required: true,
+        },
+      ],
+    },
+  } as ServiceResponse<GetCaptureStatusResponse>;
+};
+
+export const updateCaptureAnswers: UpdateCaptureAnswers = async ({ ...metadata }) => {
+  return (await updateCustomer(metadata, true, '')) as ServiceResponse<Capture>;
+};
+
+function processAccount(account: AccountData): Customer {
   const { id, email, full_name: fullName, metadata, created_at: createdAt } = account;
   const regDate = new Date(createdAt * 1000).toLocaleString();
 
   return {
     id: id.toString(),
     email,
     fullName,
-    firstName: metadata?.first_name as string,
-    lastName: metadata?.last_name as string,
+    metadata,
+    firstName: metadata?.firstName as string,
```

Comment:
these will be removed after the parent branch is merged. already updated there to be only fullName.